### ISSUE TITLE
openjdk: Java vulnerable to CVE-2022-21449; Patch available

### ISSUE THREAD

**Issue by gregod** [NixOS/nixpkgs#169633]
## Issue description

The  openjdk / jdk versions from version 15 onward are vulnerable to high severity [CVE-2022-21449](https://nvd.nist.gov/view/vuln/detail?vulnId=CVE-2022-21449)

I have verified that this is the case for the versions included in nixpkgs using the following POC using openjdk 17.

```
import java.security.*;
class POC {  
    public static void main(String args[]) throws Exception {  
        KeyPair keyPair = KeyPairGenerator.getInstance("EC").generateKeyPair();
        byte[] blankSignature = new byte[64];
        Signature sig = Signature.getInstance("SHA256WithECDSAInP1363Format");
        sig.initVerify(keyPair.getPublic());
        sig.update("Hello, world!".getBytes());
        boolean validSig = sig.verify(blankSignature);
        if (validSig) {
            System.out.println("Vulnerable to CVE-2022-21449: Blank signature accepted");  
        } else {
            System.out.println("Not vulnerable");  
        }
    }  
}  
// adapted from https://connect2id.com/blog/cve-2022-21449
```

## Suggested Solutions

Versions containing a fix have been released (for jdk17 in 17.0.3, nixpkgs master is at 17.0.1 )

Alternatively, the necessary patch (https://github.com/openjdk/jdk/commit/e2f8ce9c3ff4518e070960bafa70ba780746aa5c),  is very small and could be  backported via fetchpatch to all versions >15. 

## Notify maintainers
@edwtjo

**Comment by hqurve** [comment]
edwtjo appears to be inactive; doubt he would see this

@asbachb

**Comment by asbachb** [comment]
I think there is already a PR.

**Comment by hqurve** [comment]
😅 Should have checked before pinging.

Fixed by https://github.com/NixOS/nixpkgs/pull/169581

**Comment by gregod** [comment]
Thanks

**Comment by infinisil** [comment]
PR with the patch isn't merged yet

**Comment by greizgh** [comment]
PR is now merged and patches were also backported in #170455.

**Comment by rtimush** [comment]
Still probably open on darwin (#171269)
